### PR TITLE
Add osm tags classification

### PR DESCRIPTION
**Pull request by Santonia27**
As of now the OSM landuse data was used to classify assets. Assets falling outside the landuse polygons (BF) were removed.

Improvements:

1. OSM data is firstly classified by OSM "building" key and then updated by "landuse" values if available.  If there were empty cells or Nan values they will be updated with OSM amenity data if existing.
2.  Add Boolean to function keep_all; True= reclassify all Nan values as residential, False: remove nan values (optional)

### CODE REVIEW

**Comment by Santonia27** [comment]
@dalmijn we would like to merge this into main as this is not a floodadapt feature but a general improvement in hydromt_fiat. 
We should discuss in the next standup how we deal with these branches now.

File: hydromt_fiat/data/damage_functions/flooding/README_JRC_OSM_mapping.xlsx
Comment (panosatha) [inline hydromt_fiat/data/damage_functions/flooding/README_JRC_OSM_mapping.xlsx:1]: I usually prefer read_me files to be .txt files so they are easily readable without excel.

File: hydromt_fiat/data/damage_functions/flooding/JRC_OSM_mapping.xlsx
Comment (panosatha) [inline hydromt_fiat/data/damage_functions/flooding/JRC_OSM_mapping.xlsx:1]: I am generally not a big fun of xlsx files, and would prefer to have csv files instead, but maybe this more of a personal thing.

**Review by panosatha — COMMENTED** [review]
some minore comments. I tested in 2-3 cases and it seems to work fine!

File: hydromt_fiat/api/data_types.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ class ExposureBuildingsSettings(BaseModel):
     damage_unit: str 
     country: str = None
     bf_conversion: bool = False
+    occupancy_keep_all: bool = True
```
[the remark this answers is not in the document]
Comment (Santonia27) [reply hydromt_fiat/api/data_types.py:80]: yes nice idea, I like that

File: hydromt_fiat/data/damage_functions/flooding/JRC_OSM_mapping.xlsx
Comment (Santonia27) [reply hydromt_fiat/data/damage_functions/flooding/JRC_OSM_mapping.xlsx:1]: I agree, as its more interoperable and accessible for other users also. I just saw that there are some xlsx files in the repo already so I wasn't sure what the best practice here is. I can change it to csv

File: hydromt_fiat/data/damage_functions/flooding/README_JRC_OSM_mapping.xlsx
Comment (Santonia27) [reply hydromt_fiat/data/damage_functions/flooding/README_JRC_OSM_mapping.xlsx:1]: I updated it to a text file.

**Review by panosatha — APPROVED** [review]
Looks good!

**Pull request merged**